packages/overlay/imperative-api.md

```diff
@@ -0,0 +1,95 @@
+    (options?: OverlayOptions)
+);
+```
+
+`Overlay.open()` is an asynchronous method that returns an `<sp-overlay>` element that wraps the `HTMLElement` provided as the `overlayElement`. While this process will set the `<sp-overlay>` element to `open`, consumers of this API will need to choose where to append this element to the DOM in order for the content to be made available in the browser.
+
+```js
+(async () => {
+    const content = document.querySelector('#content');
+    const options = {
+        offset: 0,
+        placement: 'bottom',
+        trigger: document.querySelector('#trigger'),
+        type: 'auto',
+    };
+    const overlay = await Overlay.open(content, options);
+    document.body.append(overlay);
+})();
+```
+
+Keep in mind that a changing DOM tree is likely to alter the relationship between existing content. Without proper care this can negatively effect the CSS that you have applied to existing content. DOM events and DOM selection methods can also perform differently than expected as the tree shape changes.
+
+## OverlayOptions
+
+When leveraging `Overlay.open()`, you can provide an optional second argument of `OverlayOptions`, with the following type:
+
+```ts
+type OverlayOptions = {
+    delayed?: boolean;
+    notImmediatelyCloseable?: boolean;
+    offset?: number | [number, number];
+    placement?: Placement;
+    receivesFocus?: 'auto' | 'true' | 'false';
+    trigger?: HTMLElement | VirtualTrigger;
+    type?: 'modal' | 'page' | 'hint' | 'auto' | 'manual';
+};
+```
+
+### delayed
+
+An Overlay that is `delayed` will wait until a warm-up period of 1000ms has completed before opening. Once the warmup period has completed, all subsequent Overlays will open immediately. When no Overlays are opened, a cooldown period of 1000ms will begin. Once the cooldown has completed, the next Overlay to be opened will be subject to the warm-up period if provided that option.
+
+### notImmediatelyCloseable
+
+When an Overlay is `notImmediatelyCloseable` that means that the first interaction that would lead to the closure of the Overlay in question will be ignored. This is useful when working with non-"click" mouse interactions, like `contextmenu`, where the trigger event (e.g. `contextmenu`) occurs _before_ an event that would close said overlay (e.g. `pointerup`).
+
+### offset
+
+The `offset` property accepts either a single number, to define the offset of the Overlay along the main axis from the trigger, or 2-tuple, to define the offset along the main axis and the cross axis. This option has no effect when there is no trigger element.
+
+### placement
+
+A `placement` of `"auto" | "auto-start" | "auto-end" | "top" | "bottom" | "right" | "left" | "top-start" | "top-end" | "bottom-start" | "bottom-end" | "right-start" | "right-end" | "left-start" | "left-end"` will instruct the Overlay where to place itself in relationship to the trigger element.
```

Comment:
Just a side note - can you pls either link to the explanation of placement options or add some details here? I am personally interested in more details about those "auto*" options. Thx!

Reply:
`auto` may not be a part of the default API anymore, but I'll get a link to https://floating-ui.com/docs/computePosition#placement somewhere in there.